### PR TITLE
Fixes #322 & #324: Hide InfoBox Content when no results available

### LINKED ISSUE #322: Hide InfoBox and the InfoBox columns when there are no results.
- There are various queries for which all the columns inside the InfoBox are empty and it shows a blank box. One should hide it in this case.

screenshot 90

- Also in some case one of the Top Twiterrers, Top Mentions, Top Hashtags are not there in the list but the Heading is displayed. For ex., in this case *Top Mentions*.
screenshot 91

### PR DESCRIPTION
**Pull request by rishiraj824**
**Changes proposed in this pull request**
- Hides InfoBox Columns when there is no data available
- Hides InfoBox when no column is available

**Screenshots** 
- **Hides Respective Columns when no data available for eg, here Top Mentions**
screenshot 98

- **Hides the InfoBox when no data is available at all**
screenshot 99

**Link to live demo:** 
[rishiraj.co/loklak_search](http://rishiraj.co/loklak_search)

**Closes #322 & #324**

### CODE REVIEW

**Comment by Achint08** [comment]
@rishiraj824 Use '===' for type checking instead of '=='.

File: src/app/feed/info-box/info-box.component.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
-<div class="info-box">
+<div class="info-box" *ngIf="areTopMentionsAvailable || areTopTwitterersAvailable || areTopHashtagsAvailable || areFrequencyDataAvailable">
 	<div class="row">
-		<div class="col-sm-12 col-xs-12 col-md-6" in-viewport (inViewport)="inviewtwitters($event)">
+		<div class="col-sm-12 col-xs-12 col-md-6"  in-viewport (inViewport)="inviewtwitters($event)" *ngIf="areTopTwitterersAvailable">
```
Comment (uttu357) [inline src/app/feed/info-box/info-box.component.html]: The space before in-viewport is not needed...

File: src/app/feed/info-box/info-box.component.ts [hunk 2]
```diff
@@ -47,39 +51,79 @@ export class InfoBoxComponent implements OnInit, OnChanges {
...
 	}
 	sortTwiterers(statistics) {
 		let sortable = [];
-		for (let s in statistics.screen_name) {
-			sortable.push([s, statistics.screen_name[s]]);
+		if((statistics && statistics.screen_name)!=undefined && (statistics && Object.keys(statistics.screen_name).length)!=null) {
+			for (let s in statistics.screen_name) {
+				sortable.push([s, statistics.screen_name[s]]);
+			}
+				sortable.sort(function (a, b) {
```
Comment (uttu357) [inline src/app/feed/info-box/info-box.component.ts]: indentation

File: src/app/feed/info-box/info-box.component.ts [hunk 3]
```diff
@@ -47,39 +51,79 @@ export class InfoBoxComponent implements OnInit, OnChanges {
...
-		for (let s in statistics.mentions) {
+		if((statistics && statistics.mentions)!=undefined && (statistics && Object.keys(statistics.mentions).length)!=null) {
+			for (let s in statistics.mentions) {
 			sortable.push([s, statistics.mentions[s]]);
-		}
-		sortable.sort(function (a, b) {
+			}
+
+			sortable.sort(function (a, b) {
 			return b[1] - a[1];
```
Comment (uttu357) [inline src/app/feed/info-box/info-box.component.ts]: indentation

Comment (rishiraj824) [inline src/app/feed/info-box/info-box.component.ts]: will make the changes 👍

File: src/app/feed/info-box/info-box.component.html (hunk 1 above)
Comment (rishiraj824) [inline src/app/feed/info-box/info-box.component.html]: will make the changes 👍

**Comment by rishiraj824** [comment]
@Achint08 & @UtkarshGpta I have made the changes. Please review :)

File: src/app/feed/info-box/info-box.component.html (hunk 1 above)
Comment (uttu357) [inline src/app/feed/info-box/info-box.component.html]: This one is left.. :)

**Comment by uttu357** [comment]
The demo is not loading anymore please check.. :)

**Comment by rishiraj824** [comment]
@uttu357 Fixed the indentation errors. The link is up now. :)

**Comment by rishiraj824** [comment]
@hemantjadon Please check now :)

File: src/app/feed/info-box/info-box.component.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
-<div class="info-box">
+<div class="info-box" *ngIf="areTopMentionsAvailable || areTopTwitterersAvailable || areTopHashtagsAvailable || areFrequencyDataAvailable">
```
Comment (hemantjadon) [inline src/app/feed/info-box/info-box.component.html]: Please make these in separate lines like
```html
<div class="info-box" 
    *ngIf="areTopMentionsAvailable || 
                areTopTwitterersAvailable || 
                areTopHashtagsAvailable || 
                areFrequencyDataAvailable">
```

**Comment by rishiraj824** [comment]
@hemantjadon  Here you go :)

**Comment by mariobehling** [comment]
Seems like http://loklak.net is down after this merge. Any ideas?

**Pull request merged**